File: deck/seed/code/native/kotlin/runtime/regex.kt
object regex {
    fun matches(pattern: String, text: String): Boolean = Regex(pattern).containsMatchIn(text)
    fun replace(pattern: String, text: String, replacement: String): String = Regex(pattern).replace(text, replacement)
    fun find(pattern: String, text: String): String = Regex(pattern).find(text)?.value ?: ""
}

File: deck/seed/code/native/swift/runtime/regex.swift
import Foundation

enum regex {
    static func matches(_ pattern: String, _ text: String) -> Bool {
        guard let re = try? NSRegularExpression(pattern: pattern) else { return false }
        return re.firstMatch(in: text, range: NSRange(text.startIndex..., in: text)) != nil
    }
    static func replace(_ pattern: String, _ text: String, _ replacement: String) -> String {
        guard let re = try? NSRegularExpression(pattern: pattern) else { return text }
        return re.stringByReplacingMatches(in: text, range: NSRange(text.startIndex..., in: text), withTemplate: replacement)
    }
    static func find(_ pattern: String, _ text: String) -> String {
        guard let re = try? NSRegularExpression(pattern: pattern),
              let m = re.firstMatch(in: text, range: NSRange(text.startIndex..., in: text)),
              let r = Range(m.range, in: text) else { return "" }
        return String(text[r])
    }
}
